### PR TITLE
Format display numbers

### PR DESCRIPTION
**Pull request by yCobanoglu**

### CODE REVIEW

File: src/fragmentarium/ui/display/Transliteration.tsx [hunk 1]
```diff
@@ -109,32 +112,89 @@ class LineAccumulator {
     )
   }
 }
+function DisplayLineNumberRange({ start, end }: LineNumberRange): JSX.Element {
```
Comment (jlaasonen) [inline src/fragmentarium/ui/display/Transliteration.tsx:115]: It is probably not a good idea to use a model type as prop type. Better to pass the line number as a single prop. You can still use destructuring e.g. `{range: { start, end}}: {range: LineNumberRange}`.

File: src/fragmentarium/ui/display/Transliteration.tsx [hunk 2]
```diff
@@ -109,32 +112,89 @@ class LineAccumulator {
...
+  const endSuffixMod = end.suffixModifier ? end.suffixModifier : ''
+  const result = `(${startPrefixMod}${start.number}${startPrime}${startSuffixMod}-${endPrefixMod}${end.number}${endPrime}${endSuffixMod})`
+  return (
+    <span>
+      <sup>{result}</sup>
+    </span>
+  )
+}
 
+function DisplayLineNumber({ hasPrime, number }: LineNumber): JSX.Element {
```
Comment (jlaasonen) [inline src/fragmentarium/ui/display/Transliteration.tsx:130]: It is probably not a good idea to use a model type as prop type. See the comment on`DisplayLineNumberRange`.

File: src/fragmentarium/ui/display/Transliteration.tsx [hunk 3]
```diff
@@ -109,32 +112,89 @@ class LineAccumulator {
...
+  return React.createElement(
+    container,
+    { className: classNames([`Transliteration__${textLine.type}`]) },
+    [
+      React.createElement(lineTypeComponent, {
+        key: 0,
+        ...textLine.lineNumber
+      }),
+      ...helperDisplayLineTokens(line.content)
+    ]
```
Comment (jlaasonen) [inline src/fragmentarium/ui/display/Transliteration.tsx:162]: There is no need to have an array here, you can just have two elements: the line number and the content.

Creating the line number should be extracted to own component.

File: src/fragmentarium/ui/display/Transliteration.tsx [hunk 4]
```diff
@@ -109,32 +112,89 @@ class LineAccumulator {
...
-        } else if (isDocumentOrientedGloss(token)) {
-          token.side === 'LEFT' ? acc.openGloss() : acc.closeGloss()
-        } else {
-          acc.pushToken(token)
-        }
-        return acc
-      }, new LineAccumulator()).result
+      <span key="prefix">{line.prefix}</span>,
+      ...helperDisplayLineTokens(line.content)
     ]
```
Comment (jlaasonen) [inline src/fragmentarium/ui/display/Transliteration.tsx:178]: There is no need to have an array here, you can just have two elements: the prefix and the content.

File: src/fragmentarium/ui/display/Transliteration.tsx [hunk 5]
```diff
@@ -188,26 +249,37 @@ const lineComponents: ReadonlyMap<
   ['CompositeAtLine', DisplayDollarAndAtLine]
 ])
 
+const rulingsToNumber: ReadonlyMap<string, number> = new Map([
+  ['SINGLE', 1],
+  ['DOUBLE', 2],
+  ['TRIPLE', 3]
+])
+
```
Comment (jlaasonen) [inline src/fragmentarium/ui/display/Transliteration.tsx:257]: Why is this here?

File: src/fragmentarium/ui/display/Transliteration.tsx [hunk 6]
```diff
@@ -188,26 +249,37 @@ const lineComponents: ReadonlyMap<
...
+    })
   )
 }
+function DisplayEachRuling({
+  rulingType
+}: {
+  rulingType: string
+}): JSX.Element {
+  return <div className={rulingType} />
+}
```
Comment (jlaasonen) [inline src/fragmentarium/ui/display/Transliteration.tsx:282]: Why?

**Comment by yCobanoglu** [comment]
[Format line numbers for display](https://trello.com/c/gQh6uleV/349-format-line-numbers-for-display)

File: src/fragmentarium/ui/display/LineAccumulator.tsx [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    } else if (isDocumentOrientedGloss(token)) {
+      token.side === 'LEFT' ? acc.openGloss() : acc.closeGloss()
+    } else {
+      acc.pushToken(token)
+    }
+    return acc
+  }, new LineAccumulator()).result
+}
+
+export default function DisplayLineTokens({
```
Comment (jlaasonen) [inline src/fragmentarium/ui/display/LineAccumulator.tsx:120]: `DisplayLineTokens` is the default export. The file should be named accordingly.

File: src/fragmentarium/ui/display/LineAccumulator.tsx [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+  content
+}: {
+  content: ReadonlyArray<Token>
+}): any {
+  const tokens = accumulateTokens(content)
+  return tokens.map((token: ReactNode) => {
+    return token as JSX.Element
+  })
```
Comment (jlaasonen) [inline src/fragmentarium/ui/display/LineAccumulator.tsx:128]: This is an ugly way to ignore TS error. The map does not do anything and should be removed.

File: src/fragmentarium/ui/display/LineAccumulator.tsx [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+    }
+    return acc
+  }, new LineAccumulator()).result
+}
+
+export default function DisplayLineTokens({
+  content
+}: {
+  content: ReadonlyArray<Token>
+}): any {
```
Comment (jlaasonen) [inline src/fragmentarium/ui/display/LineAccumulator.tsx:124]: `any` type should be avoided. [Use a correct return type for function component.](https://www.typescriptlang.org/docs/handbook/jsx.html#function-component)

File: src/fragmentarium/ui/display/Transliteration.tsx [hunk 10]
```diff
@@ -1,117 +1,58 @@
...
-
-  closeGloss(): void {
-    this.inGloss = false
-  }
-
-  private requireSeparator(token: Token): boolean {
-    const noEnclosure = !isCloseEnclosure(token) && !this.enclosureOpened
-    return this.result.length === 0 || noEnclosure
-  }
+const lineNumberTypeToComponent = {
```
Comment (jlaasonen) [inline src/fragmentarium/ui/display/Transliteration.tsx:34]: `formatLineNumber` and `formatLineNumberRange` are not components. The name should be changed accordingly,

File: src/fragmentarium/ui/display/Transliteration.tsx [hunk 11]
```diff
@@ -1,117 +1,58 @@
...
-          modifiers={[this.language]}
-        />
-      )
-    )
+function DisplayLinePrefix({ line }: { line: Line }): JSX.Element {
+  if (line.type == 'TextLine') {
+    const textLine = line as TextLine
+    return <sup>{displayLineNumber(textLine.lineNumber)}</sup>
+  } else {
+    return <span key="prefix">{line.prefix}</span>
```
Comment (jlaasonen) [inline src/fragmentarium/ui/display/Transliteration.tsx:49]: The `key` should not be needed here.

File: src/fragmentarium/ui/display/Transliteration.tsx [hunk 12]
```diff
@@ -1,117 +1,58 @@
...
-  private requireSeparator(token: Token): boolean {
-    const noEnclosure = !isCloseEnclosure(token) && !this.enclosureOpened
-    return this.result.length === 0 || noEnclosure
-  }
+const lineNumberTypeToComponent = {
+  LineNumber: formatLineNumber,
+  LineNumberRange: formatLineNumberRange
+}
+function displayLineNumber(lineNumber: LineNumber | LineNumberRange): string {
+  const lineNumberComponent =
```
Comment (jlaasonen) [inline src/fragmentarium/ui/display/Transliteration.tsx:39]: See the comment above.

File: src/fragmentarium/ui/display/Transliteration.tsx [hunk 13]
```diff
@@ -1,117 +1,58 @@
...
-    </span>
-  )
+function formatLineNumber(lineNumber: LineNumber): string {
+  return `(${lineNumberToString(lineNumber)})`
 }
 
-function isCloseEnclosure(token: Token): boolean {
-  return isEnclosure(token) && ['CENTER', 'RIGHT'].includes(token.side)
+function formatLineNumberRange({ start, end }: LineNumberRange): string {
+  return `(${lineNumberToString(start)}-${lineNumberToString(end)})`
```
Comment (jlaasonen) [inline src/fragmentarium/ui/display/Transliteration.tsx:20]: The parentheses are added in two places. The could be along the `sup` below.

File: src/fragmentarium/ui/display/Transliteration.tsx [hunk 14]
```diff
@@ -1,117 +1,58 @@
...
-          <WordSeparator modifiers={[this.language]} />
-        </sup>
-      ) : (
-        <WordSeparator
-          key={`${target.length}-separator`}
-          modifiers={[this.language]}
-        />
-      )
-    )
+function DisplayLinePrefix({ line }: { line: Line }): JSX.Element {
```
Comment (jlaasonen) [inline src/fragmentarium/ui/display/Transliteration.tsx:44]: You could move this along with the line number functions to own file.

**Comment by jlaasonen** [comment]
Moved to #20
